### PR TITLE
refactor(multisig-prover)!: use CrossChainID in ConstructProof

### PR DESCRIPTION
**Pull request by cjcobb23**
https://axelarnetwork.atlassian.net/browse/AXE-2781

### CODE REVIEW

File: contracts/multisig-prover/src/execute.rs [hunk 1]
```diff
@@ -20,10 +20,16 @@ use crate::{
 pub fn construct_proof(
     deps: DepsMut,
     env: Env,
-    message_ids: Vec<String>,
+    message_ids: Vec<CrossChainId>,
 ) -> Result<Response, ContractError> {
     let config = CONFIG.load(deps.storage)?;
-    let batch_id = BatchID::new(&message_ids, None);
+    let batch_id = BatchID::new(
```
Comment (cgorenflo) [inline contracts/multisig-prover/src/execute.rs:26]: can you rename it to `BatchId`?

Comment (cgorenflo) [inline contracts/multisig-prover/src/execute.rs:26]: the constructor of BatchId is a bit strange. Why take a slice reference just to clone it into a new vector and sort it? I suggest changing the constructor to accept ownership of `Vec<CrossChainId>`

Comment (cgorenflo) [reply contracts/multisig-prover/src/execute.rs:26]: furthermore, I don't understand why it accepts a worker set. That case is never used

Comment (cjcobb23) [reply contracts/multisig-prover/src/execute.rs]: I changed the constructor to just accept a slice reference of CrossChainId, and removed the clone. We convert to string right away anyways, so we don't need to take ownership.

The worker set case is actually used by `CommandBatchBuilder::build`

File: contracts/multisig-prover/src/types.rs [hunk 2]
```diff
@@ -35,42 +36,45 @@ pub struct Command {
...
 
-impl BatchID {
-    pub fn new(message_ids: &[String], new_worker_set: Option<WorkerSet>) -> BatchID {
-        let mut message_ids = message_ids.to_vec();
+impl BatchId {
+    pub fn new(message_ids: &[CrossChainId], new_worker_set: Option<WorkerSet>) -> BatchId {
+        let mut message_ids = message_ids
+            .iter()
+            .map(|id| id.to_string())
+            .collect::<Vec<String>>();
```
Comment (fish-sammy) [inline contracts/multisig-prover/src/types.rs:77]:
```suggestion
            .collect::<Vec<_>>();
```

**Review by fish-sammy — APPROVED** [review]

**Pull request merged**